pkg/synchromanager/clustersynchro/resource_negotiator.go

```diff
@@ -103,6 +103,10 @@ func (negotiator *ResourceNegotiator) negotiatorVersions(resource schema.GroupRe
 	isLegacyResource := resourcescheme.LegacyResourceScheme.IsGroupRegistered(resource.Group)
 	if !isLegacyResource {
 		mapper, supportedVersions = negotiator.customResourceController.GetRESTMappingAndVersions(resource)
+		if mapper == nil {
+			klog.InfoS("Skip resource sync", "cluster", negotiator.name, "resource", resource, "reason", "No matches for resource")
+			return nil, nil, isLegacyResource
+		}
```

Comment:
`LegacyResourceScheme` represents the resources supported by the current kubernetes code dependency.
If `isLegacyResource` equals false, it does not necessarily mean that it is a custom resources, or it could be a new resource added to the cluster that is higher than the dependency version.

For such a resource, we need to go to the restmapper to get it, but isLegacyResource is false.

Here the code omits to determine whether it is a custom resource, but to simplify the code by whether it is possible to get mapping about the custom resource, and does not add comments, which may lead to some misunderstandings

Reply:
Ok, got it.